OrbitCore/LinuxTracingSession.h

```diff
@@ -0,0 +1,49 @@
+#ifndef ORBIT_CORE_LINUX_TRACING_SESSION_H_
+#define ORBIT_CORE_LINUX_TRACING_SESSION_H_
+
+#include "ContextSwitch.h"
+#include "EventBuffer.h"
+#include "LinuxCallstackEvent.h"
+#include "ScopeTimer.h"
+
+#include "absl/synchronization/mutex.h"
+
+// This class stores information about tracing session
+// and provides thread-safe access and record functions
+class LinuxTracingSession {
+ public:
+  LinuxTracingSession() = default;
+  LinuxTracingSession(const LinuxTracingSession&) = delete;
+  LinuxTracingSession& operator=(const LinuxTracingSession&) = delete;
+
+  void RecordContextSwitch(ContextSwitch&& context_switch);
+  void RecordTimer(Timer&& timer);
+  void RecordCallstack(LinuxCallstackEvent&& event);
+  void RecordHashedCallstack(CallstackEvent&& event);
+
+  // These move the content of corresponding buffer to
+  // the output vector. They return true if the buffer
+  // is not empty.
+  bool ReadAllContextSwitches(std::vector<ContextSwitch>* buffer);
+  bool ReadAllTimers(std::vector<Timer>* buffer);
+  bool ReadAllCallstacks(std::vector<LinuxCallstackEvent>* buffer);
+  bool ReadAllHashedCallstacks(std::vector<CallstackEvent>* buffer);
+
+  void Reset();
+
+ private:
+  // buffering data to send large messages instead of small ones
```

Comment:
Capital letter and full stop for the comment? This seems to be a practice consistently followed in the style guide. Same for all the other comments in this PR.